### PR TITLE
[doc] Generate the release notes' table of contents

### PR DESCRIPTION
**Pull request by oowekyala**
Manual maintenance was getting old imo

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">No java rules are changed!</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by jsotuyod** [comment]
The `do-release.sh` script resets the release notes after a release. I'm updating it before merge.

File: docs/pages/release_notes.md [hunk 1]
```diff
@@ -10,13 +10,7 @@ The PMD team is pleased to announce PMD {{ site.pmd.version }}.
...
 This is a {{ site.pmd.release_type }} release.
 
-### Table Of Contents
-
-* [New and noteworthy](#new-and-noteworthy)
-    *   [New Rules](#new-rules)
-* [Fixed Issues](#fixed-issues)
-* [API Changes](#api-changes)
-* [External Contributions](#external-contributions)
+{% tocmaker is_release_notes_processor %}
```
Comment (jsotuyod) [inline docs/pages/release_notes.md]: what's this `is_release_notes_processor`?

Comment (oowekyala) [reply docs/pages/release_notes.md]: The toc is generated only if that variable is true, so when pushing to github or sourceforge, but not when jekyll renders the site (in which case Jekyll already generates a toc).

**Pull request merged**